policy/policy_file.go

```diff
@@ -867,18 +940,25 @@ func DeleteAllPolicyFiles(homePath string) error {
 		org := dirInfo.Name()
 		glog.V(5).Infof("Deleting policies from directory %v", org)
 
-		// Remove the org directory.
 		pDir := homePath + "/" + org
-		if err := os.RemoveAll(pDir); err != nil {
-			glog.Errorf("Error removing policy directory %v, error: %v", pDir, err)
+		if !patternBasedOnly {
+			// Remove the org directory.
+			if err := os.RemoveAll(pDir); err != nil {
+				glog.Errorf("Error removing policy directory %v, error: %v", pDir, err)
+			}
+		} else {
+			// remove policy files that are pattern based
+			if err := DeletePolicyFilesForOrg(homePath, org, true); err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
 // This is an internal function used to find all policy files in the policy directory. Files that don't end in
 // .policy are ignored. Directories are also ignored.
-func getPolicyFiles(homePath string) ([]os.FileInfo, error) {
+func GetPolicyFiles(homePath string) ([]os.FileInfo, error) {
```

Comment:
I would rather not make this function available outside the policy package. It looks like it doesnt need to be capitalized.

Reply:
yes, my mistake. Forgot to switch it back.